iceoryx_dds/test/moduletests/test_iox_to_dds.cpp

```diff
@@ -168,144 +168,147 @@ TEST_F(Iceoryx2DDSGatewayTest, ImmediatelyConnectsCreatedDataWritersForDiscovere
+//     auto mockSubscriber = createMockIceoryxTerminal(testService);
+//     auto mockWriter = createMockDDSTerminal(testService);
+//     {
+//         InSequence seq;
+//         EXPECT_CALL(*mockSubscriber, hasNewChunks()).WillOnce(Return(true));
+//         EXPECT_CALL(*mockSubscriber, getChunk(_))
+//             .WillOnce(DoAll(SetArgPointee<0>(mockChunk.chunkHeader()), Return(true)));
+//         EXPECT_CALL(*mockWriter, write(_, _)).Times(1);
+//         EXPECT_CALL(*mockSubscriber, releaseChunk(_)).Times(1);
+//         EXPECT_CALL(*mockSubscriber, hasNewChunks())
+//             .WillRepeatedly(Return(false)); // No more chunks after the first one
+//     }
+
+//     stageMockIceoryxTerminal(std::move(mockSubscriber));
+//     stageMockDDSTerminal(std::move(mockWriter));
+
+//     auto testChannel = channelFactory(testService);
+//     TestGateway gw{};
+
+//     // === Test
+//     gw.forward(testChannel.value());
+// }
+
+// TEST_F(Iceoryx2DDSGatewayTest, DestroysCorrespondingSubscriberWhenAPublisherStopsOffering)
+// {
+//     // === Setup
+//     auto testService = iox::capro::ServiceDescription({"Radar", "Front-Right", "Reflections"});
+
+//     // Subscribers
+//     auto firstCreatedSubscriber = createMockIceoryxTerminal(testService);
+//     auto secondCreatedSubscriber = createMockIceoryxTerminal(testService);
+//     {
+//         InSequence seq;
+//         EXPECT_CALL(*firstCreatedSubscriber, subscribe(_)).Times(1);
+//         EXPECT_CALL(*secondCreatedSubscriber, subscribe(_)).Times(1);
+//     }
+
+//     stageMockIceoryxTerminal(std::move(firstCreatedSubscriber));
+//     stageMockIceoryxTerminal(std::move(secondCreatedSubscriber));
+
+//     // Messages
+//     auto offerMsg = iox::capro::CaproMessage(iox::capro::CaproMessageType::OFFER, testService);
+//     offerMsg.m_subType = iox::capro::CaproMessageSubType::EVENT;
+//     auto stopOfferMsg = iox::capro::CaproMessage(iox::capro::CaproMessageType::STOP_OFFER, testService);
+//     stopOfferMsg.m_subType = iox::capro::CaproMessageSubType::EVENT;
+
+//     // Get the test channels here as we need to use them in expectations
+//     auto testChannelOne = channelFactory(testService);
+//     auto testChannelTwo = channelFactory(testService);
+
+//     TestGateway gw{};
+//     EXPECT_CALL(gw, findChannel(_))
+//         .WillOnce(Return(iox::cxx::nullopt_t()))
+//         .WillOnce(
+//             Return(iox::cxx::make_optional<iox::gw::Channel<MockSubscriber,
+//             MockDataWriter>>(testChannelOne.value())))
+//         .WillOnce(Return(iox::cxx::nullopt_t()));
+//     EXPECT_CALL(gw, addChannel(_)).WillOnce(Return(testChannelOne)).WillOnce(Return(testChannelTwo));
+//     EXPECT_CALL(gw, discardChannel(_)).WillOnce(Return(iox::cxx::success<>()));
+
+//     // === Test
+//     gw.discover(offerMsg);
+//     gw.discover(stopOfferMsg); // first subscriber must be deleted here
+//     gw.discover(offerMsg);
+// }
```

Comment:
Please either remove the code or fix it. Maybe @ithier can support?

Reply:
@ithier is working on it in #376.